### PR TITLE
Fix the filename typo detection edge cases. Closes #1532 and some other bugs

### LINKED ISSUE #1532: Filenames of empty files are not checked
If a file has no content, `codespell --check-filenames` doesn't detect misspelled words in the filename:
```
$ codespell --version
1.18.dev0
$ touch abandonned
$ codespell --check-filenames
$ echo "foo" > abandonned
$ codespell --check-filenames
./abandonned: abandonned ==> abandoned
```

### PR DESCRIPTION
**Pull request by peternewman**
To fix #1532

### CODE REVIEW

File: codespell_lib/_codespell.py [hunk 1]
```diff
@@ -457,9 +457,6 @@ def parse_file(filename, colors, summary, misspellings, exclude_lines,
         f = sys.stdin
         lines = f.readlines()
     else:
-        # ignore binary files
-        if not os.path.isfile(filename):
-            return 0
```
Comment (peternewman) [inline codespell_lib/_codespell.py:462]: No reason we shouldn't check their names for typos first.

File: codespell_lib/_codespell.py [hunk 2]
```diff
@@ -494,15 +491,19 @@ def parse_file(filename, colors, summary, misspellings, exclude_lines,
...
+        # ignore irregular files
+        if not os.path.isfile(filename):
+            return bad_count
+
         text = is_text_file(filename)
         if not text:
             if not options.quiet_level & QuietLevels.BINARY_FILE:
                 print("WARNING: Binary file: %s" % filename, file=sys.stderr)
-            return 0
+            return bad_count
```
Comment (peternewman) [inline codespell_lib/_codespell.py:502]: We should still return any filename typos found

File: codespell_lib/_codespell.py [hunk 3]
```diff
@@ -494,15 +491,19 @@ def parse_file(filename, colors, summary, misspellings, exclude_lines,
...
         try:
             lines, encoding = file_opener.open(filename)
         except Exception:
-            return 0
+            return bad_count
```
Comment (peternewman) [inline codespell_lib/_codespell.py:506]: Likewise.

File: codespell_lib/_codespell.py [hunk 4]
```diff
@@ -722,8 +723,6 @@ def main(*args):
                     fname = os.path.join(root, file_)
                     if glob_match.match(fname):  # skip paths
                         continue
-                    if not os.path.isfile(fname) or not os.path.getsize(fname):
-                        continue
```
Comment (peternewman) [inline codespell_lib/_codespell.py:726]: We might as well handle this in parse_file so we can deal with all the filename checking in one place.

File: codespell_lib/tests/test_basic.py [hunk 5]
```diff
@@ -270,7 +270,16 @@ def test_ignore(tmpdir):
 
 def test_check_filename(tmpdir):
     """Test filename check."""
+    # Irregular file (!isfile())
+    fifodir = str(tmpdir)
+    os.mkfifo(op.join(fifodir, 'abandonned'))
+    assert cs.main('-f', fifodir) == 1
```
Comment (peternewman) [inline codespell_lib/tests/test_basic.py:276]: mkfifo is only on Unix, do we run on Windows? I guess we need to skip this test on those platforms somehow?

**Comment by larsoner** [comment]
Makes sense to me!

**Pull request merged**